js_modules/dagit/packages/core/src/ticks/EvaluateSchedule.tsx

```diff
@@ -0,0 +1,400 @@
+  Icon,
+  Menu,
+  MenuItem,
+  Mono,
+  NonIdealState,
+  Popover,
+  Spinner,
+  Subheading,
+  Tag,
+} from '@dagster-io/ui';
+import React from 'react';
+import styled from 'styled-components/macro';
+
+import {PYTHON_ERROR_FRAGMENT} from '../app/PythonErrorFragment';
+import {PythonErrorInfo} from '../app/PythonErrorInfo';
+import {TimezoneContext} from '../app/time/TimezoneContext';
+import {timestampToString} from '../app/time/timestampToString';
+import {useViewport} from '../gantt/useViewport';
+import {testId} from '../testing/testId';
+import {repoAddressToSelector} from '../workspace/repoAddressToSelector';
+import {RepoAddress} from '../workspace/types';
+
+import {RunRequestTable} from './DryRunRequestTable';
+import {
+  GetScheduleQuery,
+  GetScheduleQueryVariables,
+  ScheduleDryRunMutation,
+  ScheduleDryRunMutationVariables,
+} from './types/EvaluateSchedule.types';
+
+const locale = navigator.language;
+
+export const EvaluateSchedule: React.FC<{
+  repoAddress: RepoAddress;
+  name: string;
+  onClose: () => void;
+  isOpen: boolean;
+  jobName: string;
+}> = ({repoAddress, name, isOpen, onClose, jobName}) => {
+  const [_selectedTimestamp, setSelectedTimestamp] = React.useState<{ts: number; label: string}>();
+  const {data} = useQuery<GetScheduleQuery, GetScheduleQueryVariables>(GET_SCHEDULE_QUERY, {
+    variables: {
+      scheduleSelector: {
+        repositoryLocationName: repoAddress.location,
+        repositoryName: repoAddress.name,
+        scheduleName: name,
+      },
+    },
+  });
+  const [userTimezone] = React.useContext(TimezoneContext);
+  const [isTickSelectionOpen, setIsTickSelectionOpen] = React.useState<boolean>(false);
+  const selectedTimestampRef = React.useRef<{ts: number; label: string} | null>(null);
+  const {viewport, containerProps} = useViewport();
+  const [shouldEvaluate, setShouldEvaluate] = React.useState(false);
+  const content = React.useMemo(() => {
+    if (shouldEvaluate) {
+      return (
+        <EvaluateScheduleContent
+          repoAddress={repoAddress}
+          name={name}
+          timestamp={selectedTimestampRef.current!.ts}
+          jobName={jobName}
+        />
+      );
+    }
+    if (!data) {
+      return (
+        <Box padding={{vertical: 48}} flex={{alignItems: 'center', justifyContent: 'center'}}>
+          <Spinner purpose="page" />
+        </Box>
+      );
+    }
+    if (data.scheduleOrError.__typename === 'PythonError') {
+      return <div />;
+    }
+    if (data.scheduleOrError.__typename === 'ScheduleNotFoundError') {
+      return <div />;
+    }
+    const timestamps = data.scheduleOrError.potentialTickTimestamps.map((ts) => ({
+      ts,
+      label: timestampToString({
+        timestamp: {unix: ts},
+        locale,
+        timezone: userTimezone,
+        timeFormat: {
+          showTimezone: true,
+        },
+      }),
+    }));
+    selectedTimestampRef.current = _selectedTimestamp || timestamps[0];
+    return (
+      <SelectWrapper>
```

Comment:
This seems like a standard component that should be part of our library - is this written from scratch to get the `data-testid` props in there? Seems like it'd be best to move it to it's own component so the `isTickSelectionOpen` state doesn't have to pollute the top-level modal.

Reply:
This is just to add some custom CSS to override blueprint's CSS to make the Select button take the full width.